### PR TITLE
New Addon: Remove Autocomplete from the Search Bar

### PR DESCRIPTION
**Pull request by BroJac5246**
See #4774

Creating this PR so that we don't merge from my master branch.

Let me know if I did something wrong.

### CODE REVIEW

File: addons/remove-search-bar-autocomplete/addon.json
Comment (Weredime) [inline addons/remove-search-bar-autocomplete/addon.json:10]: Append a `/` to the end of users URIs
```suggestion
  "credits": [
    {
      "name": "jbthepig",
      "link": "https://scratch.mit.edu/users/jbthepig/"
    }
  ],
```

**Review by Weredime — COMMENTED** [review]
I think you should use `addon.tab.waitForElement` as the search bar can be removed by going in to the editor.

**Comment by Weredime** [comment]
Resolves #2983

**Comment by Weredime** [comment]
@BroJac5246 Here's some code that possibly would work, with [`addon.tab.waitForElement`](https://scratchaddons.com/docs/reference/addon-api/addon.tab/#using-addontabwaitforelement), if you need a starting point.
```js
export default async function ({ addon, global, console }) {
  // Detects whether the current page is using scratch-www or scratchr2.
  // The search bars have different selectors depending on this, so we need to figure out which selector to use.
  const SEARCH_BAR_SELECTOR = addon.tab.clientVersion === "scratch-www" ? "#frc-q-1088" : "#search-input";
  let searchBar = null;
  addon.self.addEventListener("disabled", () => {
    if (searchBar) searchBar.autocomplete = "on";
  });
  addon.self.addEventListener("reenabled", () => {
    if (searchBar) searchBar.autocomplete = "off";
  });
  while (true) {
    // Wait for the current search bar to load.
    searchBar = await addon.tab.waitForElement(SEARCH_BAR_SELECTOR, {
      markAsSeen: true,
      reduxCondition: (state) => (state.scratchGui ? state.scratchGui.mode.isPlayerOnly : true),
    });
    // Check if this addon is disabled; because the end-user
    // could have disabled the addon while they were in the editor.
    if (!addon.self.disabled) {
      searchBar.autocomplete = "off";
    }
  }
}
```

**Comment by BroJac5246** [comment]
> @BroJac5246 Here's some code that possibly would work, with [`addon.tab.waitForElement`](https://scratchaddons.com/docs/reference/addon-api/addon.tab/#using-addontabwaitforelement), if you need a starting point.

Thank you, I implemented this using the code you provided. It appears to work, but I'd appreciate if it could be verified, since search bar autocomplete is a finicky thing.
<sub>Sorry it took so long, I've been super busy recently.</sub>

**Comment by BroJac5246** [comment]
Weredime's code got me thinking, who do I need to add to the credits for the addon?

**Comment by Weredime** [comment]
> Weredime's code got me thinking, who do I need to add to the credits for the addon?

Keep the credits as the same, I just refactored your code :P

**Review by Weredime — APPROVED** [review]

**Comment by BroJac5246** [comment]
Just want to point out that [TheColaber gave approval](https://github.com/ScratchAddons/ScratchAddons/pull/4774#pullrequestreview-1037509860) on the old pull request

**Comment by BroJac5246** [comment]
Just want to point out that [TheColaber gave approval](https://github.com/ScratchAddons/ScratchAddons/pull/4774#pullrequestreview-1037509860) on the old pull request

**Comment by Weredime** [comment]
> Just want to point out that [TheColaber gave approval](https://github.com/ScratchAddons/ScratchAddons/pull/4774#pullrequestreview-1037509860) on the old pull request

cc @TheColaber could you please check and approve again?

**Comment by BroJac5246** [comment]
Is there anything else I need to do to get this merged, or just wait?

**Comment by TheColaber** [comment]
@WorldLanguages can you confirm whether this addon is rejected or not?

**Comment by WorldLanguages** [comment]
@TheColaber See https://github.com/ScratchAddons/ScratchAddons/pull/4774#issuecomment-1179569880

I remember this being suggested by SA users, so it can be added as a non-featured "small" addon.

**Review by WorldLanguages — APPROVED** [review]

**Comment by WorldLanguages** [comment]
Can a Firefox user test this before merging?

**Comment by Samq64** [comment]
> Can a Firefox user test this before merging?

The addon has no effect and there are no errors in the console. Is that the expected behavior?

**Comment by WorldLanguages** [comment]
> The addon has no effect and there are no errors in the console. Is that the expected behavior?

With this addon disabled, does the Scratch search bar autocomplete for you?

**Comment by Samq64** [comment]
> > The addon has no effect and there are no errors in the console. Is that the expected behavior?
> 
> With this addon disabled, does the Scratch search bar autocomplete for you?

No, I don't think so.

**Comment by WorldLanguages** [comment]
I get something like this in Chromium, with this addon disabled:

image

**Comment by Weredime** [comment]
> I get something like this in Chromium, with this addon disabled:

Can't repro on Firefox

**Comment by Samq64** [comment]
> I get something like this in Chromium, with this addon disabled:

I don't get that on Chromium or Firefox with the addon disabled.

**Comment by Weredime** [comment]
addon works on firefox

**Comment by WorldLanguages** [comment]
> I don't get that on Chromium or Firefox with the addon disabled.

I think the browser tries to autocomplete with anything that was previously submitted to any `<input name="q">`
But if your browser doesn't do that for some reason, it's out of scope :P It does for me and other people

**Pull request merged**
